lang/typescript/src/extended-address/splitAddress1.ts

```diff
@@ -1,25 +1,52 @@
 import type {Address} from '../types/address';
-import {splitAddressField} from '../utils/address-fields';
-import {getRegionConfig, getConcatenationRules} from '../utils/regions';
+import {
+  RESERVED_DELIMITER,
+  splitAddressField,
+  regexSplitAddressField,
+} from '../utils/address-fields';
+import {
+  getRegionConfig,
+  getConcatenationRules,
+  getAddress1Regex,
+} from '../utils/regions';
 
 /**
- * Parse a concatenated address1 string based on the region specified by
- * country code
+ * Splits an address string into sub-fields using a reserved delimiter.
+ * Optionally provides a fallback mechanism to parse the address using
+ * a regex when the delimiter is absent, which should be used with caution
+ * as it may not provide accurate results.
+ *
  * @param countryCode 2-letter country code string
- * @param concatenatedAddress Combined address1 string
+ * @param address Combined address1 string
+ * @param experimentalRegexFallback Flag to attempt regex parsing as a fallback mechanism
  * @returns Partial Address object containing parsed address fields or null if
  * the region does not define an extended address format
  */
 export function splitAddress1(
   countryCode: string,
-  concatenatedAddress: string,
+  address: string,
+  experimentalRegexFallback = false,
```

Comment:
Open to other names. I considered `tryRegexFallback` as it's more concise and suggests that the splitting may not be successful

Reply:
`try` as a prefix works, although I think `experimental` feels more "dangerous" which seems like it's what we want to convey. Up to you, I'm not too concerned with the specific prefix though as long as it seems to make sense when reading. Just don't use `useRegexFallback` as that prefix has a specific use case in react and could confuse consumers.